### PR TITLE
Fix regression of enable-crypto-mdebug on Windows

### PR DESCRIPTION
**Pull request by t8m**
Some of the non-default options that enable more
code to be built need to be enabled in one of the
Windows builds to avoid regressions.

### CODE REVIEW

**Comment by t8m** [comment]
This fixes a regression from #22127

**Review by levitte — APPROVED** [review]

**Comment by t8m** [comment]
@levitte Please reapprove.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -237,7 +237,7 @@ jobs:
     - name: modprobe tls
       run: sudo modprobe tls
     - name: config
-      run: ./config --banner=Configured --strict-warnings no-ec enable-ssl-trace enable-zlib enable-zlib-dynamic enable-crypto-mdebug enable-crypto-mdebug-backtrace enable-egd enable-ktls enable-fips no-threads && perl configdata.pm --dump
+      run: ./config --banner=Configured --strict-warnings no-ec enable-ssl-trace enable-zlib enable-zlib-dynamic enable-crypto-mdebug enable-egd enable-ktls enable-fips no-threads && perl configdata.pm --dump
```
Comment (t8m) [inline .github/workflows/ci.yml:240]: enable-crypto-mdebug-backtrace is now just a no-op so there is no point in having it there.

File: crypto/mem.c [hunk 2]
```diff
@@ -122,12 +122,13 @@ static void parseit(void)
 }
 
 /*
- * Windows doesn't have random(), but it has rand()
+ * Windows doesn't have random() and srandom(), but it has rand() and srand().
```
Comment (tom-cosgrove-arm) [inline crypto/mem.c:125]: (trivial, please don't iterate just for this) should more properly be "Windows doesn't have random() or srandom()"

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM

**Review by paulidale — APPROVED** [review]

**Pull request closed without merging**